Data Science with Python : Complete Overview of Regression

yashh18 opened the pull request:
<hr>

## Description 📜

Documentation contribution for Overview of Regression
Fixes : #735 

<hr>

## Type of change 📝

<!----Please delete the hashtag from the correct option----->


- [x] Documentation (Content Creation in the form of codes or tutorials)


<hr>

## Domain of Contribution 📊

<!----Please delete the hashtag from your domain----->


- [x] Datascience with Python


<hr>

## Checklist ✅

<!----Please delete options that are not relevant.And in order to tick the check box just but x inside them for example [x] like this----->

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

<!----Please delete options that are not relevant.And in order to tick the check box just but x inside them for example [x] like this----->

## Screenshots / Gif (Optional) 📸

<hr>

yashh18 commented:
@kareem-negm @Yash621 @prathimacode-hub Please review my PR!

Yash621 reviewed (commented):
Firstly make the readme file title bold ,after that make a separate folder for regression analysis under the folder of algorithms ,and place your regression overview file there.
secondly in your explanation you have only talked about linear regression and I can't find enough explanation on polynomial regression ,if you are mentioning a algorithm ,try to give a code example also for them.
and lastly try to include a brief overview of other types of regression like logistics ,lasso,ridge etc under the heading "some other imp types of regression algorithms".No need to give a code example for them just definition and formula.

yashh18 commented:
@Yash621 okay cool
So do i have to delete the whole folder and re-upload it or what's the process of editing it?

prathimacode-hub commented:
That's upto you. You can do it afresh or do it it in your existing PR. @yashh18

prathimacode-hub reviewed (changes requested):
Changes to be done 👍 

There is no sub repo of Documentation in Machine Learning. It should go in "Tutorials" folder
All images should be in "Images" folder
The project folder should be titled as "Overview of Regression"
The README.md file should be elaborate, it's not sufficient. It only has outline of it. You can add features, why its used, application, advantages, conclusion.
The readme file should be titled as "README.md"
In .ipynb file, you can add few more code snippets to represent the topics given. Also at end, overall summary of the topic.

Kindly work on it. @yashh18

Kareem-negm commented on Datascience_With_Python/Machine Learning/Overview_of_Regression/Documentation/overview_of_regression.ipynb:
```diff
@@ -0,0 +1,297 @@
+        "Example: Suppose there is a marketing company A, who does various advertisement every year and get sales on that. The below list shows the advertisement made by the company in the last 5 years and the corresponding sales:\r\n",
+        "\r\n",
+        "\r\n",
+        "![image](images\\regression-analysis-in-machine-learning.png)\r\n",
```
please fix this image URL too

Kareem-negm commented on Datascience_With_Python/Machine Learning/Overview_of_Regression/Documentation/overview_of_regression.ipynb:
```diff
@@ -0,0 +1,297 @@
+        "\r\n",
+        "Bias is a deviation induced to the line equation $y = mx$ for the predictions we make. We need to tune the bias to vary the position of the line that can fit best for the given data. \r\n",
+        "\r\n",
+        "![linreg](images\\linear.jpeg)\r\n",
```
this too

Kareem-negm commented on Datascience_With_Python/Machine Learning/Overview_of_Regression/Documentation/overview_of_regression.ipynb:
```diff
@@ -0,0 +1,297 @@
+        "\r\n",
+        "If n=1, the polynomial equation is said to be a linear equation.\r\n",
+        "\r\n",
+        "![pol](images\\pr7.png)"
```
this too

yashh18 commented:
@prathimacode-hub Please review it once again !

prathimacode-hub reviewed (changes requested):
Changes to be done :

Your README needs more detailing. This is a vast topic atleast a fair introduction related to this topic is needed. 
You don't have to add all your files in Documentation folder. Instead it can be uploaded in "Overview of Regression" as we have separate folders of Audio and Video
The project file name should be in snake case : overview_of_regression.ipynb
All the images should be in "Images" folder.

Kindly work on it. @yashh18

yashh18 commented:
@prathimacode-hub I have deleted the documentation folder and added the files in Overview of regression, changed the .ipynb file to snakecase, added all the images in the required folder and covered most of the information in both readme and jupyter file. 
So please review it once again !

prathimacode-hub reviewed (approved):
PR approved. 👍@yashh18

yashh18 commented:
@prathimacode-hub thank you ✨
